src/bam/collapse_reads_to_fragments/calc_consensus.rs

```diff
@@ -79,6 +87,41 @@ impl<'a> CalcOverlappingConsensus<'a> {
     fn overlap(&self) -> usize {
         self.overlap
     }
+    fn build_consensus_strand(
+        &self,
+        consensus_strand: &mut Vec<u8>,
+        ref_base: u8,
+        base_pos: usize,
+    ) {
+        let mut strands = HashSet::new();
+        let fwd_end_pos = self.recs1()[0].len();
+        let rev_start_pos = fwd_end_pos - self.overlap();
+        if base_pos < fwd_end_pos {
+            self.recs1().iter().for_each(|rec| {
+                if rec.base(base_pos) == ref_base {
+                    match rec.is_reverse() {
+                        true => strands.insert(b'-'),
+                        false => strands.insert(b'+'),
+                    };
+                }
+            });
+        }
+        if base_pos >= rev_start_pos {
+            let rev_base_pos = base_pos - rev_start_pos;
+            self.recs2().iter().for_each(|rec| {
+                if rec.base(rev_base_pos) == ref_base {
+                    match rec.is_reverse() {
+                        true => strands.insert(b'-'),
+                        false => strands.insert(b'+'),
+                    };
+                }
+            });
+        }
+        match strands.len() == 1 {
+            true => consensus_strand.push(strands.take(&(b'-')).unwrap_or(b'+')),
```

Comment:
I don't like the way how the single element from within the HashSet is accessed, but I did not find another way like poping or indexing. As it is guaranteed to contain just a single element is there any way to returning it without this `take` and `unwrap_or` thingy?

Reply:
See the enum suggestion above. This avoids sets entirely.